editor/groups_editor.cpp

```diff
@@ -32,714 +32,679 @@
+		hbc->add_child(add_group_name);
+
+		HBoxContainer *hbc_desc = memnew(HBoxContainer);
+
+		Label *label_description = memnew(Label(TTR("Description:")));
+		label_description->set_custom_minimum_size(Size2(100 * EDSCALE, 1));
+		hbc_desc->add_child(label_description);
+
+		add_group_description = memnew(LineEdit);
+		add_group_description->set_custom_minimum_size(Size2(300 * EDSCALE, 1));
+		add_group_description->set_editable(false);
+		hbc_desc->add_child(add_group_description);
+
+		global_group_button = memnew(CheckButton);
+		global_group_button->set_text(TTR("Global"));
+		global_group_button->connect("toggled", callable_mp(add_group_description, &LineEdit::set_editable));
+		hbc->add_child(global_group_button);
+
+		add_error = memnew(Label);
+
+		vbc->add_child(hbc);
+		vbc->add_child(hbc_desc);
+		vbc->add_child(add_error);
+		add_child(add_group_dialog);
 	}
-};
+	add_group_name->clear();
+	add_group_description->clear();
 
-void GroupsEditor::update_tree() {
-	tree->clear();
+	global_group_button->set_pressed(false);
 
-	if (!node) {
+	_check_add("");
+
+	add_group_dialog->popup_centered();
+	add_group_name->grab_focus();
+}
+
+void GroupsEditor::_show_rename_group_dialog() {
+	if (!rename_group_dialog) {
+		rename_group_dialog = memnew(ConfirmationDialog);
+		rename_group_dialog->set_title(TTR("Rename Group"));
+
+		VBoxContainer *vbc = memnew(VBoxContainer);
+		rename_group_dialog->add_child(vbc);
+
+		HBoxContainer *hbc = memnew(HBoxContainer);
+		hbc->add_child(memnew(Label(TTR("Name:"))));
+
+		rename_group = memnew(LineEdit);
+		rename_group->set_custom_minimum_size(Size2(300 * EDSCALE, 1));
+		rename_group->connect("text_changed", callable_mp(this, &GroupsEditor::_check_rename));
+		hbc->add_child(rename_group);
+
+		rename_error = memnew(Label);
+		vbc->add_child(hbc);
+		vbc->add_child(rename_error);
+
+		rename_group_dialog->connect("confirmed", callable_mp(this, &GroupsEditor::_confirm_rename));
+		add_child(rename_group_dialog);
+	}
+
+	TreeItem *ti = tree->get_selected();
+	if (!ti) {
 		return;
 	}
 
-	List<Node::GroupInfo> groups;
-	node->get_groups(&groups);
-	groups.sort_custom<_GroupInfoComparator>();
+	String name = ti->get_meta("__name");
 
-	TreeItem *root = tree->create_item();
+	rename_group->set_text(name);
+	rename_group_dialog->set_meta("__name", name);
 
-	for (const GroupInfo &gi : groups) {
-		if (!gi.persistent) {
-			continue;
-		}
+	_check_rename(name);
 
-		Node *n = node;
-		bool can_be_deleted = true;
+	rename_group_dialog->popup_centered();
+	rename_group->select_all();
+	rename_group->grab_focus();
+}
 
-		while (n) {
-			Ref<SceneState> ss = (n == EditorNode::get_singleton()->get_edited_scene()) ? n->get_scene_inherited_state() : n->get_scene_instance_state();
+void GroupsEditor::_show_remove_group_dialog() {
+	if (!remove_group_dialog) {
+		remove_group_dialog = memnew(ConfirmationDialog);
+		remove_group_dialog->set_title(TTR("Please confirm..."));
```

Comment:
```suggestion
```

ConfirmationDialog title is "Please Confirm..." by default.

Reply:
Are you suggesting removing the title from the dialog? Delete dialogs that I saw have the title "Please confirm..." :)

Reply:
No, because `ConfirmationDialog.title` is "Please Confirm..." by default. So if this line is removed the title won't change.

![image](https://user-images.githubusercontent.com/67974470/169058417-f70dab05-1120-4039-ade2-a1746b299a3f.png)